internals/overlord/logstate/opentelemetry/opentelemetry.go

```diff
@@ -126,11 +128,12 @@ type Client struct {
 	resourceAttributes map[string][]keyValue
 }
 
-func NewClient(options *ClientOptions) *Client {
+func NewClient(target *plan.LogTarget, options *ClientOptions) *Client {
```

Comment:
I think we should revert this change (in both `loki` and `opentelemetry`). I think avoiding the dependency on the `plan` package is a good thing (even if it means the "duplicated" `TargetName` and `Location` fields) -- there's no reason these log forwarders should need to know about or depend on the Pebble plan.

So let's add `TargetName` and `Location` back and remove the *target* argument from `NewClient` in both packages.